src/dos/drive_local.cpp

```diff
@@ -56,22 +56,50 @@ bool localDrive::FileCreate(DOS_File * * file,char * name,uint16_t /*attributes*
 	/* Test if file exists (so we need to truncate it). don't add to dirCache then */
 	bool existing_file = false;
 	
-	FILE * test = fopen_wrap(temp_name,"rb+");
+	FILE* test = fopen_wrap(temp_name, "rb");
 	if (test) {
 		fclose(test);
-		existing_file=true;
+		existing_file = true;
+	}
+
+	FILE* fp = nullptr;
 
+#ifdef WIN32
+	DWORD win_attribs = attributes ? (DWORD)attributes : FILE_ATTRIBUTE_NORMAL;
+	HANDLE win_fhandle = CreateFile(newname,
+	                                GENERIC_READ | GENERIC_WRITE,
+	                                0,
+	                                nullptr,
+	                                CREATE_ALWAYS,
+	                                win_attribs,
+	                                nullptr);
+	if (win_fhandle != INVALID_HANDLE_VALUE) {
+		int fd = _open_osfhandle((intptr_t)win_fhandle, _O_RDWR | _O_BINARY);
+		fp = _fdopen(fd, "wb+");
 	}
-	
-	FILE * hand = fopen_wrap(temp_name,"wb+");
-	if (!hand) {
+#else
+	if (attributes & (DOS_ATTR_SYSTEM | DOS_ATTR_HIDDEN)) {
+		LOG_WARNING("FILESYSTEM: Application attempted to create '%s' with system or"
+		            " hidden attributes, which is ignored for local drives",
+		            newname);
+	}
+
+	int unix_perms = (attributes & DOS_ATTR_READ_ONLY) ? 0444 : 0666;
+	int fd         = open(newname, O_CREAT | O_RDWR | O_TRUNC, unix_perms);
+	if (fd != -1) {
+		fp = fdopen(fd, "wb+");
+	}
+#endif
+
+	if (!fp) {
 		LOG_MSG("Warning: file creation failed: %s",newname);
+		DOS_SetError(ERROR_ACCESS_DENIED);
```

Comment:
You've got a compile error here:  `../src/dos/drive_local.cpp:96:16: error: ‘ERROR_ACCESS_DENIED’ was not declared in this scope; did you mean ‘DOSERR_ACCESS_DENIED’?`

Reply:
Whoops.